File: webapi/src/Application/Common/Http/RequestItemFilter.cs
namespace NetClock.Application.Common.Http
{
    public class RequestItemFilter
    {
        public string PropertyName { get; set; }

        public string RelationalOperator { get; set; }

        public string Value { get; set; }

        public string LogicalOperator { get; set; }
    }
}

File: webapp/src/app/models/request-data/request-item-filter.ts
import { LogicalOperator } from './logical-operator';
import { RelationalOperator } from './relational-operator';

export class RequestItemFilter {
  propertyName: string;
  relationalOperator: RelationalOperator;
  value: string;
  logicalOperator: LogicalOperator;

  constructor(propertyName: string, operator: RelationalOperator, value: string, concat = LogicalOperator.none) {
    this.propertyName = propertyName;
    this.relationalOperator = operator;
    this.value = value;
    this.logicalOperator = concat;
  }
}
